modules/ximgproc/src/paillou_filter.cpp

```diff
@@ -0,0 +1,449 @@
+#include "precomp.hpp"
+#include "opencv2/highgui.hpp"
+#include <math.h>
+#include <vector>
+#include <iostream>
+
+namespace cv {
+namespace ximgproc {
+
+/*
+If you use this code please cite this @cite paillou1997detecting
+Detecting step edges in noisy SAR images: a new linear operator  IEEE Transactions on Geoscience and Remote Sensing  (Volume:35 ,  Issue: 1 )  1997
+*/
+
+
+class ParallelGradientPaillouYCols: public ParallelLoopBody
+{
+private:
+    Mat &img;
+    Mat &im1;
```

Comment:
I think im1 name is bit confusing. Maybe dst is better (as below in other classes)